modules/git/command.go

```diff
@@ -110,24 +110,47 @@ func (c *Command) RunInDirTimeoutEnvFullPipeline(env []string, timeout time.Dura
 // RunInDirTimeoutEnvFullPipelineFunc executes the command in given directory with given timeout,
 // it pipes stdout and stderr to given io.Writer and passes in an io.Reader as stdin. Between cmd.Start and cmd.Wait the passed in function is run.
 func (c *Command) RunInDirTimeoutEnvFullPipelineFunc(env []string, timeout time.Duration, dir string, stdout, stderr io.Writer, stdin io.Reader, fn func(context.Context, context.CancelFunc) error) error {
-	if timeout == -1 {
-		timeout = defaultCommandExecutionTimeout
+	return c.RunWithContext(&RunContext{
+		Env:        env,
+		Timeout:    timeout,
+		Dir:        dir,
+		Stdout:     stdout,
+		Stderr:     stderr,
+		Stdin:      stdin,
+		CancelFunc: fn,
+	})
+}
+
+// RunContext represents parameters to run the command
+type RunContext struct {
+	Env            []string
+	Timeout        time.Duration
+	Dir            string
+	Stdout, Stderr io.Writer
+	Stdin          io.Reader
+	CancelFunc     func(context.Context, context.CancelFunc) error
```

Comment:
This isn't a good name for this.

It's not a cancel function - it doesn't match the context.CancelFunc interface - and it's actually meant to do work.

```suggestion
	PipelineFunc     func(context.Context, context.CancelFunc) error
```

would probably be better